dialects/sqlite-3-18/src/main/kotlin/app/cash/sqldelight/dialects/sqlite_3_18/grammar/mixins/PragmaFunctionValidatorMixin.kt

```diff
@@ -0,0 +1,43 @@
+package app.cash.sqldelight.dialects.sqlite_3_18.grammar.mixins
+
+import com.alecstrong.sql.psi.core.SqlAnnotationHolder
+import com.alecstrong.sql.psi.core.psi.SqlCompositeElement
+import com.alecstrong.sql.psi.core.psi.SqlTypes
+import com.alecstrong.sql.psi.core.psi.impl.SqlPragmaNameImpl
+import com.intellij.lang.ASTNode
+import com.intellij.psi.tree.TokenSet
+
+internal abstract class PragmaFunctionValidatorMixin(node: ASTNode) : SqlPragmaNameImpl(node) {
```

Comment:
because this is being used in the from clause i expect it will also need to return correct types

you can look at the json mixin for inspiration for doing something similar, since it also adds table valued functions for the sqlite json extension:

https://github.com/cashapp/sqldelight/blob/master/dialects/sqlite/json-module/src/main/kotlin/app/cash/sqldelight/dialects/sqlite/json/module/grammar/json.bnf#L23-L30